File: java/webflux-security-example/src/main/java/com/van1164/webfluxsecurityexample/security/JwtAuthenticationFilter.java
package com.van1164.webfluxsecurityexample.security;

import com.van1164.webfluxsecurityexample.user_info.JwtTokenProvider;
import lombok.AllArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.server.reactive.ServerHttpRequest;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.stereotype.Component;
import org.springframework.util.StringUtils;
import org.springframework.web.server.ServerWebExchange;
import org.springframework.web.server.WebFilter;
import org.springframework.web.server.WebFilterChain;
import reactor.core.publisher.Mono;

@Component
@Slf4j
@AllArgsConstructor
public class JwtAuthenticationFilter implements WebFilter {


    JwtTokenProvider jwtTokenProvider;

    @Override
    public Mono<Void> filter(ServerWebExchange exchange, WebFilterChain chain) {
        log.info("Enter jwt filter");
        String token = resolveToken(exchange.getRequest());
        if (token != null && jwtTokenProvider.validateToken(token)) {
            Authentication authentication = jwtTokenProvider.getAuthentication(token);
            SecurityContextHolder.getContext().setAuthentication(authentication);
            log.info("doFilterChain:$authentication");
        }
        return chain.filter(exchange);
    }

    private String resolveToken(ServerHttpRequest request) {
        String bearerToken = request.getHeaders().getFirst("Authorization");
        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer")) {
            return bearerToken.substring(7);
        }
        return null;
    }

}

File: kotlin/webflux-security-example/src/main/kotlin/com/van1164/webfluxsecurityexample/security/JwtAuthenticationFilter.kt
package com.van1164.webfluxsecurityexample.security

import com.van1164.webfluxsecurityexample.logger
import com.van1164.webfluxsecurityexample.user_info.JwtTokenProvider
import org.springframework.http.server.reactive.ServerHttpRequest
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.stereotype.Component
import org.springframework.util.StringUtils
import org.springframework.web.server.ServerWebExchange
import org.springframework.web.server.WebFilter
import org.springframework.web.server.WebFilterChain
import reactor.core.publisher.Mono

@Component
class JwtAuthenticationFilter(
    private val jwtTokenProvider: JwtTokenProvider
) : WebFilter {

    override fun filter(exchange: ServerWebExchange, chain: WebFilterChain): Mono<Void> {
        logger.info {"Enter jwt filter"}
        val token = resolveToken(exchange.request)
        if (token != null && jwtTokenProvider.validateToken(token)) {
            val authentication = jwtTokenProvider.getAuthentication(token)
            SecurityContextHolder.getContext().authentication = authentication
            logger.info("doFilterChain:$authentication")
        }
        return chain.filter(exchange)
    }

    private fun resolveToken(request : ServerHttpRequest) : String? {
        val bearerToken = request.headers["Authorization"]?.get(0)
        return if (StringUtils.hasText(bearerToken) && bearerToken!!.startsWith("Bearer")) {
            bearerToken.substring(7)
        } else {
            null
        }
    }

}